src/caf/toolkit/cost_utils.py

```diff
@@ -6,98 +6,116 @@
 import copy
 import logging
 import os
-from typing import TYPE_CHECKING, Optional
+from typing import Optional
 
 # Third Party
 import numpy as np
 import pandas as pd
-import pydantic
 
 # Local Imports
 from caf.toolkit import math_utils
 from caf.toolkit import pandas_utils as pd_utils
 
-if TYPE_CHECKING:
-    from dataclasses import dataclass  # isort:skip
-else:
-    from pydantic.dataclasses import dataclass  # isort:skip
-
 # # # CONSTANTS # # #
 LOG = logging.getLogger(__name__)
 
 
 # # # CLASSES # # #
-@dataclass(config={"arbitrary_types_allowed": True})  # type: ignore # pylint: disable=used-before-assignment, unexpected-keyword-arg
 class CostDistribution:
     """Distribution of cost values between variable bounds.
 
-    Attributes
+    Alternate constructors are available in the See Also section
+
+    Parameters
     ----------
     df:
-        The raw Pandas DataFrame containing the cost distribution data.
+        A DataFrame containing the binned cost distribution data. Must have columns
+        named: `min_col`, `max_col`, `avg_col`, `trips_col`.
 
-    min_vals:
-        Minimum values of the cost distribution bin edges.
+    min_col:
+        The name of the columns in `df` that contains the lower bin edge
+        value for each row.
 
-    max_vals:
-        Maximum values of the cost distribution bin edges.
+    max_col:
+        The name of the columns in `df` that contains the upper bin edge
+        value for each row.
 
-    bin_edges:
-        Bin edges for the cost distribution.
-
-    avg_vals:
-        Average values for each of the cost distribution bins.
+    avg_col:
+        The name of the columns in `df` that contains the centre of the bin
 
-    trip_vals:
-        Trip values for each of the cost distribution bins.
+    trips_col:
+        The name of the columns in `df` that contains the value for each
+        row.
 
-    band_share_vals:
-        Band share values for each of the cost distribution bins.
+    weighted_avg_col:
+        The name of the columns in `df` that contains the weighted average
+        value for each row. If available, this is different from `avg_col`
+        as it takes into account this distribution of values within each
+        bound when calculating averages.
 
-    weighted_avg_vals:
-        Weighted average values for each of the cost distribution bins.
+    See Also
+    --------
+    :func:`from_data`
+    :func:`from_data_no_bins`
+    :func:`from_file`
     """
 
-    df: pd.DataFrame
-
-    # Default arguments
-    min_col: str = "min"
-    max_col: str = "max"
-    avg_col: str = "ave"
-    trips_col: str = "trips"
-    weighted_avg_col: Optional[str] = "weighted_ave"
-
     # Ideas
     units: str = "km"
```

Comment:
Does this want to be a parameter that instead of being hardcoded?

Reply:
Weird hangover for development - will comment out
```suggestion
    # units: str = "km"
```